keps/sig-cli/3659-kubectl-apply-prune/README.md

```diff
@@ -627,128 +544,171 @@ In future, we may define additional “index” mechanisms here to further optim
-
-We will detect “overlapping” ApplySets where objects already have a different ApplySet label, and initially treat this
-an error. During implementation of the alpha we will explore to what extent we can
-optimize this overlap discovery, particularly in conjunction with
-server-side-apply which does not require an object read before applying.
-A richer apply tooling than kubectl does will likely establish watches
-on the objects before applying them, to monitor object health and status.
-However, this is out of scope for kubectl and thus we will likely have to
-optimize differently for kubectl.  In the worst case, we will have to fetch
-the objects before applying (with a set of label-filtered LIST requests),
-we will explore to what extent that can be amortized over other kubectl
-operations in alpha.  One interesting option may be to use the fieldManager,
-choosing a fieldManager that includes the ApplySet ID to automatically
-detect conflicts (by _not_ specifying force); we intend to explore
-how this looks in practice and whether other options present themselves.
+The `--namespace` flag will be required when using any namespaced parent, including the default Secret. Because that flag throws a mismatch error when the set contains resources with heterogeneous namespaces, this limits the scope of ApplySet-based pruning in kubectl specifically beyond what the spec proposed strictly requires. Specifically, in kubectl, ApplySets spanning multiple namespaces MUST use a cluster-scoped parent object. We believe this limitation is reasonable and encourages best practices, but we could consider relaxing this position (e.g. using the ApplySet-in-input option described above) based on user feedback. When applicable, kubectl will ensure that all "visited" namespaces (as defined in the current operational code) are named by the sum of the parent's own namespace (if any) and the `applyset.kubernetes.io/additional-namespaces` annotation.
+
+In the initial implementation, the initial apply operation will overwrite the `applyset.kubernetes.io/part-of` annotation (unless this is automatically prevented, e.g. by a conflict caused by SSA use on that label by two different tools). This effectively allows the set to thrash, which is undesirable. Before beta, we will explore whether it is possible to detect such "overlapping" ApplySets with acceptable performance, particularly in conjunction with server-side-apply, which does not require an object read before applying.
```

Comment:
Note that the 'automatic' prevention I mention here won't happen even if the user is using SSA, if the two effectively competing `apply` operations are both kubectl using the default field manager.